core/chain_makers.go

```diff
@@ -58,7 +59,11 @@ func (b *BlockGen) SetCoinbase(addr common.Address) {
 		panic("coinbase can only be set once")
 	}
 	b.header.Coinbase = addr
-	b.gasPool = new(GasPool).AddGas(b.header.GasLimit)
+	// See core/gaspool.go for details on how these gas limit values are calculated
+	b.gasPool = NewLegacyGasPool(b.config, b.header.Number, new(big.Int).SetUint64(b.header.GasLimit))
```

Comment:
Current specification does not have 2 distinct gas pools. We should consider merge them in a single one.